arduino/arduino-ide
# fix: execute the Arduino CLI without a `shell`

### Motivation


Execute the Arduino CLI process without a `shell`.
 - security reasons,
 - some users might not have the `%ComSpec%` set on Windows, and `CMD.exe` is not on the `$PATH`

Remove the obsolete CLI daemon watcher process. Before https://github.com/arduino/arduino-cli/pull/488, IDE2 required a way to stop the daemon process if the parent (backend) process crashed. However, this mechanism
is no longer necessary, as the CLI daemon process is not actually a true daemon process.

### Change description


### Other information


To verify:
 - Arduino IDE starts, and the CLI is fully functional when there is a whitespace in the path to your IDE2 installation.
 - Start IDE2. Open the task manager/activity monitor on your system. See that `arduino-cli` process is running. Kill the IDE2 process. See that the `arduino-cli` process is not running.

Closes https://github.com/arduino/arduino-ide/issues/2112
Ref: https://github.com/arduino/arduino-ide/pull/2067

### Reviewer checklist
